lib/ood_core/job/info.rb

```diff
@@ -162,6 +162,23 @@ def gpu?
         gpus.positive?
       end
 
+      # Compute the total memory being used by a job
+      # @return [Integer] total memory in use
+      def total_memory
+        # Ensure the scheduler has needed fields for computation
+        return nil unless native[:min_memory] && allocated_nodes&.any?
+
+        # Using the adapter, key off whether the job is CPU or node calculated
+        case native[:memory_per]
+        when :cpu
+          native[:min_memory] * procs
+        when :node
+          native[:min_memory] * allocated_nodes.count
+        else
+          nil
+        end
+      end
```

Comment:
I think we should just have an attribute and return it. Let the adapters figure out what to do, while this object just holds the scalar number.

This object should not be pulling from `native` because then this object needs to know implementation details of all the adapters.

Instead, when the adapters are building an Info object, they should just pass in a memory key to the constructor and let this object just return whatever the adapter set.

Reply:
Sure! Totally agree. I can remove this method and add an attr that just draws from the adapter, in this case slurm, to compute and provide the `total_memory` to the new Info#total_memory attr.